src/scanner/mod.rs

```diff
@@ -130,8 +137,33 @@ impl Scanner {
     ///
     /// Note: `self` must contain `self.ip`.
     async fn scan_socket(&self, socket: SocketAddr) -> io::Result<SocketAddr> {
-        let tries = self.tries.get();
+        if self.sudp {
+            // we try different wait times and packet types
+            // I looked for a crate to create these packets but couldn't find one
```

Comment:
You are welcome to make a sub-crate in RustScan if you  think a crate is worth it :) 

I think it's https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html , although only if you think it's worth it. It's a lot of overhead to make a sub crate